### PR TITLE
Show all markers on current view + center the map around selected marker

### PR DESCRIPTION
**Pull request by bitamar**
#25

### CODE REVIEW

File: src/js/elm-interop.js [hunk 1]
```diff
@@ -2,11 +2,6 @@
 
 var elmApp = Elm.fullscreen(Elm.Main, {selectEvent: null});
 
-// @todo: Remove this hack, that makes sure that the map will appear on first
-// load, as the subscribe to port is triggered only on the first change of
-// model, and not when it is initialized.
-elmApp.ports.selectEvent.send(null);
```
Comment (bitamar) [inline src/js/elm-interop.js]: The map seems to be working fine without this

File: src/js/elm-interop.js [hunk 2]
```diff
@@ -73,8 +77,12 @@ elmApp.ports.mapManager.subscribe(function(model) {
 /**
  * Send marker click event to Elm.
  */
-function selectMarker(markerEl, id) {
-  markerEl.on('click', function(e) {
+function selectMarker(mapEl, markerEl, id) {
+  markerEl.on('click', function(event) {
+
+    // Center the map around the selected event.
+    mapEl.panTo(event.latlng);
```
Comment (bitamar) [inline src/js/elm-interop.js]: Currently it only works when selecting a marker on the map, not when selecting on the events list.

File: src/js/elm-interop.js [hunk 3]
```diff
@@ -84,7 +92,7 @@ function selectMarker(markerEl, id) {
  */
 function addMap() {
   // Leaflet
-  var mapEl = L.map('map').setView([50, 50], 3);
```
Comment (bitamar) [inline src/js/elm-interop.js]: No need to set the default view now, as fitBounds sets it according to the available markers.

File: src/js/elm-interop.js [hunk 4]
```diff
@@ -58,14 +53,23 @@ elmApp.ports.mapManager.subscribe(function(model) {
       eventIds.splice(index, 1);
     });
 
+    //
+    if (model.leaflet.markers.length) {
+      mapEl.fitBounds(model.leaflet.markers);
+    }
+    else {
+       // Show the entire world when no markers are set.
+      mapEl.setZoom(1);
```
Comment (amitaibu) [inline src/js/elm-interop.js]: can we panTo it?

File: src/js/elm-interop.js [hunk 5]
```diff
@@ -58,14 +53,23 @@ elmApp.ports.mapManager.subscribe(function(model) {
       eventIds.splice(index, 1);
     });
 
+    //
```
Comment (amitaibu) [inline src/js/elm-interop.js]: Add comment.

File: src/js/elm-interop.js (hunk 2 above)
Comment (amitaibu) [inline src/js/elm-interop.js]: this should move out of the click

**Comment by amitaibu** [comment]
@bitamar needs a re-roll

**Comment by amitaibu** [comment]
Merged, with proper credit. Thanks!

**Pull request closed without merging**